### PR TITLE
update sliders when change scale of layer

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Add update of sliders on change scale of the layer. 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -557,6 +557,7 @@ def add_layer(self, layer: layers.Layer) -> layers.Layer:
         layer.events.data.connect(self._on_layers_change)
         layer.name = self.layers._coerce_name(layer.name, layer)
         layer.events.name.connect(self.layers._update_name)
+        layer.events.scale.connect(self._on_layers_change)
```
Comment (sofroniewn) [inline napari/components/viewer_model.py:560]: Should we add for `translate`, `rotate`, `shear` and `affine` too? We might want to rethink this long term, but if we add for one, might as well add for all

Comment (Czaki) [reply napari/components/viewer_model.py:560]: I meet this problem when update the scale. I'm not sure which are also need, so I'm waiting on such comment like Yours. 

I personally prefer to add some event which should be emitted on every change which should cause the call of function like `_on_layers_change`. It's maybe useful with also some synchronization etc. 

But looking at the suggestion to release 0.4.2 on zulip I prefer to do a fast bugfix, and go back to refactor later.

Comment (sofroniewn) [reply napari/components/viewer_model.py:560]: yup agreed, we should refactor later, i think this whole `_on_layers_change` stuff should change a lot

**Review by sofroniewn — APPROVED** [review]
This now looks good to me as a quick bug fix, but we can continue to improve later.

A small test (could just be for scale say) might be nice too. Let me know if you can do that @Czaki?

**Comment by Czaki** [comment]
@sofroniewn I add a simple test

**Review by sofroniewn — APPROVED** [review]
Love it, thanks for the test @Czaki! I will merge once tests pass. Much appreciated!!

**Pull request merged**